### PR TITLE
Re-instate BboxConnectorPatch fill

### LINKED ISSUE #8059: BboxConnectorPatch does not show facecolor
### Bug report

**Bug summary**

When adding a BboxConnectorPatch, the edges are displayed correctly but the facecolor is not shown. The face is empty.

**Code for reproduction**
```python
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.transforms import Bbox, TransformedBbox, \
     blended_transform_factory
from mpl_toolkits.axes_grid1.inset_locator import BboxPatch, BboxConnector,\
     BboxConnectorPatch
fig, ax = plt.subplots(1, 2)
fig.set_size_inches(12, 6)
fig.set_tight_layout(True)
ax[0].set_xlim(0, 1)
ax[0].set_ylim(0, 1)
ax[1].set_xlim(0, 1)
ax[1].set_ylim(0.35, 0.65)
trans1 = blended_transform_factory(ax[0].transData, ax[0].transData)
trans2 = blended_transform_factory(ax[1].transData, ax[1].transData)
bbox = Bbox.from_extents(0, 0.4, 1, 0.6)
bbox1 = TransformedBbox(bbox, trans1)
bbox2 = TransformedBbox(bbox, trans2)
p = BboxConnectorPatch(bbox1, bbox2,
                       loc1a=1, loc2a=2, loc1b=4, loc2b=3, ec="b", fc="b")
p.set_clip_on(False)
ax[0].add_patch(p)
ax[1].yaxis.tick_right()
plt.show()
```

**Actual outcome**

download

**Expected outcome**

I would expect the trapezoid to be filled in blue rather than to be empty.

**Matplotlib version**
Matplotlib version: 2.0.0
Python version: 3.4.2
Platform: Linux
Python was installed from the debian repositories and matplotlib was installed via pip.

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #8059. Also adds a simple image test for `BboxConnectorPatch`, which is currently not tested.

### CODE REVIEW

**Comment by tacaswell** [comment]
Did you check that the examples this change was introduced to fix are not re-broken?

**Comment by dstansby** [comment]
I have just checked, and all the examples mentioned in #5757 still work as expected.

File: lib/mpl_toolkits/axes_grid1/inset_locator.py [hunk 1]
```diff
@@ -312,7 +312,8 @@ def __init__(self, bbox1, bbox2, loc1, loc2=None, **kwargs):
             raise ValueError("transform should not be set")
 
         kwargs["transform"] = IdentityTransform()
-        Patch.__init__(self, fill=False, **kwargs)
+        fill = 'fc' in kwargs
```
Comment (tacaswell) [inline lib/mpl_toolkits/axes_grid1/inset_locator.py]: This needs to also check for 'facecolor' and 'color'.  Use `cbook.normalize_kwargs` to sort out the aliases.

Comment (dstansby) [inline lib/mpl_toolkits/axes_grid1/inset_locator.py]: I haven't use normalize_kwargs before; something like `kwargs = cbook.normalize_kwargs(kwargs, alias_mapping={'fc': ['facecolor', 'color']})` ?

Comment (tacaswell) [inline lib/mpl_toolkits/axes_grid1/inset_locator.py]: See https://github.com/matplotlib/matplotlib/blob/e69ea36995a76c37b928f9cab9af1161f8d36e05/lib/matplotlib/axes/_axes.py#L1406 for an example.

You can use `normalize_kwargs` to sort out `fc` vs `facecolor`, etc .  You will have to do the color / facecolor / edgecolor normalization by hand I think.

Comment (dstansby) [inline lib/mpl_toolkits/axes_grid1/inset_locator.py]: Thanks for the example, I should have fixed this now.

**Comment by Mr-Ian-Ferguson** [comment]
The latest [commit ](https://github.com/matplotlib/matplotlib/pull/8075/commits/b843f9c3ca728bc4f3e07b72111d0aefd132e2e7) should fix the issue if facecolor or color is given (or an alias) but I see a few potential problems.

1. If you look at the documentation for [BboxConnectorPatch](http://matplotlib.org/mpl_toolkits/axes_grid/api/inset_locator_api.html#mpl_toolkits.axes_grid1.inset_locator.BboxConnectorPatch) you can see fill is also a valid kwarg argument. By always setting fill in the init, if a user actually passed in the fill argument [see line 319](https://github.com/matplotlib/matplotlib/pull/8075/commits/b843f9c3ca728bc4f3e07b72111d0aefd132e2e7#diff-3067988cda8ab99e169537e73b1e063dL316) it would throw:
`TypeError: __init__() got multiple values for keyword argument 'fill'`
which does not seem like intended behaviour.
The [mark_inset](https://github.com/matplotlib/matplotlib/blob/master/lib/mpl_toolkits/axes_grid1/inset_locator.py#L585) function also suffers from this problem, which should also be able to take fill arguments [see docs](http://matplotlib.org/mpl_toolkits/axes_grid/api/inset_locator_api.html#mpl_toolkits.axes_grid1.inset_locator.mark_inset).

2. It seems like a bit of a maintenance issue to manually map the aliases on [line 316](https://github.com/matplotlib/matplotlib/pull/8075/commits/b843f9c3ca728bc4f3e07b72111d0aefd132e2e7#diff-3067988cda8ab99e169537e73b1e063dR316) if they are ever updated or change, this would then break. Instead I would recommend handleing the kwargs like in [matplotlib/axes/_axes.py Line 1971](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/axes/_axes.py#L1971) and reference the [_patch_alias_map](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/patches.py#L60) directly (I know it has the _ private identifier but this seems consistent with how this is handled elsewhere in the existing code).

Suggested Fix:
For BboxConnectorPatch I would suggest something like:
`from matplotlib.cbook import normalize_kwargs`
`import matplotlib.patches as Patches`
`.`
`.`
`.`
`kwargs = normalize_kwargs(kwargs, Patches._patch_alias_map)`
`if 'fill' in kwargs:`
`    Patch.__init__(self, **kwargs)`
`else:`
`    fill = ('facecolor' in kwargs) or ('color' in kwargs):`
`    Patch.__init__(self, fill=fill, **kwargs)`

AND

For mark_inset something like:
`kwargs = normalize_kwargs(kwargs, Patches._patch_alias_map)`
`if 'fill' in kwargs:`
`    pp = BboxPatch(rect, **kwargs)`
`else:`
`    fill = ('facecolor' in kwargs) or ('color' in kwargs):`
`    pp = BboxPatch(rect, fill=fill, **kwargs)`

That way any valid input should be handled correctly.

**Comment by dstansby** [comment]
@Mr-Ian-Ferguson could you put your suggested changes in a new pull request?

**Comment by tacaswell** [comment]
@Mr-Ian-Ferguson @dstansby What is the status of this?

**Comment by dstansby** [comment]
I don't plan to pick this up, but thought I'd leave the PR open for reference.

**Comment by Mr-Ian-Ferguson** [comment]
Sorry I've been without internet for a while. I probably won't be picking this up again, just something I noticed in passing while looking for a different issue. That said I stand by my above suggestions and your welcome to use as you see fit; I just don't have time to do proper testing right now or do a full PR.

**Comment by dstansby** [comment]
I still have no plans to pick this up, so going to close.

**Pull request closed without merging**